### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by hgraca**
- Added a demo run.php file in bin/
- Added .idea to gitignore
- Added a demo box.json
- Improved the readme file to explain how to create and use new commands, as well as creating a phar file using box

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ cilex.phar
 docs/.build
 composer.lock
 .vagrant
+.idea
```
Comment (kamilsk) [inline .gitignore]: I think it is better to use [global gitignore](https://help.github.com/articles/ignoring-files/) in your development environment

Comment (bluemanos) [inline .gitignore]: Of course the line should be placed in developer global gitignore. Any files not related to a project shouldnt be ignored by a project gitignore file.

@hgraca imagine if in project will be hundreds developers using 50 different IDEs - should every developer place their own IDE configuration ignore in a project .gitignore file?

File: README.md [hunk 2]
```diff
@@ -57,6 +72,7 @@ Cilex is licensed under the MIT license.
 [2]: http://silex.sensiolabs.org
 [3]: http://cilex.github.com/get/cilex.phar
 [4]: http://cilex.github.com/documentation
+[5]: http://box-project.org/
```
Comment (bluemanos) [inline README.md]: The http://box-project.org/ domain isnt registered right now.

File: .gitignore (hunk 1 above)
Comment (hgraca) [inline .gitignore]: Damn!! Sorry guys, I completely missed @kamilsk comment last year!! (!!!)
Tkx for making a note about it @bluemanos .

**Comment by hgraca** [comment]
Cool! Thank you.

**Comment by Seldaek** [comment]
Thanks for the improvements.. I tagged 2.0.0 now using all this. Ideally though I think cilex/cilex should be installed as a dependency rather than checked out like this and then modified, but that woudl require us to provide a demo/sample application in another repository, and I don't really have time for that.

**Comment by hgraca** [comment]
Ok, nice to know the ideal direction of the project.
Maybe I will dive into it soon then, whenever I have the time for it. :)

**Pull request merged**